build.sh

```diff
@@ -175,8 +179,28 @@ if [ "$MACHINE" == "ncar" ]; then
   fi
 fi
 
-
-MOM6_src_files=${MOM_ROOT}/{config_src/infra/FMS2,config_src/memory/${MEMORY_MODE},config_src/drivers/solo_driver,pkg/CVMix-src/src/shared,pkg/GSW-Fortran/modules,../MARBL/src,config_src/external,src/{*,*/*}}/
+# comma-separated list of files in src/framework that are needed to build libinfra (for FMS2, at least)
+MOM6_infra_framework_deps=`printf "%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s" MOM_string_functions.F90 \
+                                                                                    MOM_io.F90 \
+                                                                                    MOM_array_transform.F90 \
+                                                                                    MOM_domains.F90 \
+                                                                                    MOM_error_handler.F90 \
+                                                                                    posix.F90 \
+                                                                                    MOM_file_parser.F90 \
+                                                                                    MOM_coms.F90 \
+                                                                                    MOM_document.F90 \
+                                                                                    MOM_cpu_clock.F90 \
+                                                                                    MOM_unit_scaling.F90 \
+                                                                                    MOM_dyn_horgrid.F90 \
+                                                                                    MOM_hor_index.F90 \
+                                                                                    MOM_ensemble_manager.F90 \
+                                                                                    MOM_io_file.F90 \
+                                                                                    MOM_netcdf.F90`
```

Comment:
Would this be a more portable option?
```suggestion
MOM6_infra_framework_deps_list=(
  "MOM_string_functions.F90"
  "MOM_io.F90"
  "MOM_array_transform.F90"
  "MOM_domains.F90"
  "MOM_error_handler.F90"
  "posix.F90"
  "MOM_file_parser.F90"
  "MOM_coms.F90"
  "MOM_document.F90"
  "MOM_cpu_clock.F90"
  "MOM_unit_scaling.F90"
  "MOM_dyn_horgrid.F90"
  "MOM_hor_index.F90"
  "MOM_ensemble_manager.F90"
  "MOM_io_file.F90"
  "MOM_netcdf.F90"
)
MOM6_infra_framework_deps = $(echo "${MOM6_infra_framework_deps_list[@]}" | sed 's/ /,/g')
```